CRM/Core/BAO/Domain.php

```diff
@@ -144,10 +144,15 @@ public function getLocationValues() {
    * @param int $id
    *
    * @return CRM_Core_DAO_Domain
+   * @throws \CRM_Core_Exception
    */
-  public static function edit($params, $id) {
+  public static function edit($params, $id): CRM_Core_DAO_Domain {
     $params['id'] = $id;
-    return self::writeRecord($params);
+    $result = self::writeRecord($params);
+    // We want to clear out any cached tokens.
+    // Editing a domain is so rare we can risk being heavy handed.
+    Civi::cache('metadata')->clear();
+    return $result;
```

Comment:
@eileenmcnaughton I think it would be better to implement this functionality using `self_hook_civicrm_post()`, because this function looks screwy and probably ought to be deprecated. From the looks of it, APIv4 probably bypasses this function so your cache clear will also get skipped.